node/pkg/governor/governor_prices.go

```diff
@@ -81,50 +104,25 @@ func (gov *ChainGovernor) PriceQuery(ctx context.Context) error {
 	}
 }
 
-// queryCoinGecko sends a query to the CoinGecko server to get the latest prices. It can
+// queryCoinGecko sends a series of of one or more queries to the CoinGecko server to get the latest prices. It can
 // return an error, but that is only used by the tool that validates the query. In the actual governor,
 // it just logs the error and we will try again next interval. If an error happens, any tokens that have
 // not been updated will be assigned their pre-configured price.
 func (gov *ChainGovernor) queryCoinGecko() error {
-	response, err := http.Get(gov.coinGeckoQuery)
-	if err != nil {
-		gov.logger.Error("cgov: failed to query coin gecko, reverting to configured prices", zap.String("query", gov.coinGeckoQuery), zap.Error(err))
-		gov.revertAllPrices()
-		return fmt.Errorf("failed to query CoinGecko")
-	}
-
-	defer func() {
-		err = response.Body.Close()
+	result := make(map[string]interface{})
+	for queryIdx, query := range gov.coinGeckoQueries {
+		thisResult, err := gov.queryCoinGeckoChunk(query)
 		if err != nil {
-			gov.logger.Error("cgov: failed to close coin gecko query")
-			// We can't safely call revertAllPrices() here because we don't know if we hold the lock or not.
-			// Also, we don't need to because the prices have already been updated / reverted by this point.
+			gov.logger.Error("cgov: coin gecko query failed", zap.Int("queryIdx", queryIdx), zap.String("query", query), zap.Error(err))
```

Comment:
Do you want to keep "cgov:" in the log messages?

Reply:
Fixed "coin gecko".

The change from "cgov: " to using a logger component will be done with a separate PR.